### ISSUE TITLE
[bug] not possible to override sub-dependency

### ISSUE THREAD

**Issue by udovenkoav1981** [conan-io/conan#9775]
### Environment Details (include every applicable attribute)
  * OS independent:
  * Compiler independent:
  * Conan version 1.40:
  * Python version 3.6.9:

### Steps to reproduce (Include if Applicable)
conan install gst-plugins-base/1.19.1@ --build=gst-plugins-base

### Logs (Executed commands with output) (Include/Attach if Applicable)
ERROR: Conflict in flex/2.6.4:
    'flex/2.6.4' requires 'm4/1.4.19' while 'bison/3.7.6' requires 'm4/1.4.18'.
    To fix this conflict you need to override the package 'm4' in your root package.

Ok, we see some probles with dependency, let's try to fix it via conanfile.txt:
[requires]
gst-plugins-base/1.19.1
m4/1.4.19

execute comand:
conan install . --build=gst-plugins-base

get same result:
ERROR: Conflict in flex/2.6.4:
    'flex/2.6.4' requires 'm4/1.4.19' while 'bison/3.7.6' requires 'm4/1.4.18'.
    To fix this conflict you need to override the package 'm4' in your root package.

not possible to  override sub-dependency, but for direct dependency - all ok.

**Comment by memsharded** [comment]
It seems this conflict comes because of ``build_requires``, not regular requires, and might not be possible to override it as a regular require.

Could you please try the following?
- Create a local profile file, like ``myprofile``
- Add there:
  ```
  [build_requires]
  bison:m4/1.4.19
  ```
- Run your ``conan install ... -pr=myprofile`` to inject that build_require

**Comment by udovenkoav1981** [comment]
i try, got same situation:

/temp$ conan install . --build=gst-plugins-base
Configuration:
[settings]
arch=x86_64
arch_build=x86_64
build_type=Release
compiler=gcc
compiler.libcxx=libstdc++11
compiler.version=7
cuda_version=10.0
os=Linux
os_build=Linux
[options]
[build_requires]
bison: m4/1.4.19
[env]

gst-plugins-base/1.19.1: Forced build from source
ERROR: Conflict in flex/2.6.4:
    'flex/2.6.4' requires 'm4/1.4.19' while 'bison/3.7.6' requires 'm4/1.4.18'.
    To fix this conflict you need to override the package 'm4' in your root package.

**Comment by udovenkoav1981** [comment]
I make a copy of recipie gst-plugins-base, and add manually self.build_requires("m4/1.4.19") to section build_requirements, and got same situation.


    def build_requirements(self):
        self.build_requires("meson/0.54.2")
        if not tools.which("pkg-config"):
            self.build_requires("pkgconf/1.7.4")
        if self.settings.os == 'Windows':
            self.build_requires("winflexbison/2.5.24")
        else:
            self.build_requires("bison/3.7.6")
            self.build_requires("flex/2.6.4")
            self.build_requires("m4/1.4.19")
        if self.options.with_introspection:
            self.build_requires("gobject-introspection/1.68.0")

how i can make any fast local fix?

**Comment by memsharded** [comment]
But you didn't use the profile, your command line is ``conan install . --build=gst-plugins-base``, and it must include ``--profile=myprofile``. Or did you changed it in your default profile?

**Comment by udovenkoav1981** [comment]
> Or did you changed it in your default profile?

Yes.

**Comment by memsharded** [comment]
I am pinging @uilianries and @SSE4 to see if there is something on the ConanCenter recipe side that could help with this.

**Comment by udovenkoav1981** [comment]
I think you need to provide possibility to  override dependency in build requires.

**Comment by SSE4** [comment]
there were some conflicts in conan-center, but should be already fixed by:
https://github.com/conan-io/conan-center-index/pull/7656
https://github.com/conan-io/conan-center-index/pull/7650

**Comment by udovenkoav1981** [comment]
Yes, conflict was fixed yesterday. But for future - will be good to have possibility to override dependency in build requires.

**Comment by uilianries** [comment]
There is a workaround using dual profile: `conan install foo/0.1@ -pr:b default -pr:h default` It will "ignore" the conflict between build_requires and requires.

**Comment by AbrilRBS** [comment]
Closing as solved in Conan 2, where it's easy to change versions of dependencies using profiles if the need arises, even for build requires
